components/device/DownloadCard.vue

```diff
@@ -0,0 +1,254 @@
+        class="download-button"
+      >
+        <template #icon>
+          <IconDownload />
+        </template>
+      </Button>
+    </div>
+    <template v-if="$slots.default">
+      <Divider />
+      <div class="changelog-title-bar">
+        <Button
+          variant="text"
+          class="expand-button"
+          :class="expand ? 'upside-down' : ''"
+          @click="() => (expand = !expand)"
+        >
+          <template #icon>
+            <IconExpandMore />
+          </template>
+        </Button>
+        <div class="changelog-title-text">
+          {{ t("deviceDownloadChangelog") }}
+        </div>
+      </div>
+      <div class="changelog">
+        <div class="changelog-main">
+          <slot />
+        </div>
+        <div class="changelog-detail" v-show="expand">
+          <slot name="detail" />
+        </div>
+      </div>
+    </template>
+  </div>
+  <div v-else class="mb-[24px]">
+    <Panel toggleable collapsed>
+      <template #header>
+        <b class="panel-header">
+          <div>
+            <span>{{ title }}</span>
+            <span v-if="debug">{{
+              t("deviceDownloadDebugVersion")
+            }}</span>
+          </div>
+          <span class="panel-header-version">
+            {{ version }}
+          </span>
+        </b>
+      </template>
+      <div class="metadata">
+        <span>{{ formattedSize }}</span>
+        <span>{{ formattedDate }}</span>
+        <CopyInline :text="sha256" type="SHA-256" />
+      </div>
+      <Button
+        as="a"
+        :href="url"
+        label="Download"
+        class="download-button mt-[8px]"
+      >
+        <template #icon>
+          <IconDownload />
+        </template>
+      </Button>
+      <div v-if="$slots.default" class="changelog">
+        <Divider />
+        <div class="changelog-title-text">
+          {{ t("deviceDownloadChangelog") }}
+        </div>
+        <div class="changelog-detail">
+          <slot />
+        </div>
+      </div>
+    </Panel>
+  </div>
+</template>
+
+<script setup>
+import { ref, computed } from "vue";
+import { useI18n } from "vue-i18n";
+import { Button, Panel, Divider } from "primevue";
+import IconDownload from "~icons/material-symbols/download";
+import IconExpandMore from "~icons/material-symbols/expand-more";
+import CopyInline from "./CopyInline.vue";
+
+const props = defineProps({
+  title: String,
+  version: String,
+  size: Number,
+  date: String, // Unix 时间戳或者 Date() 认识的格式
+  sha256: String,
+  url: String,
+  latest: Boolean,
+  debug: Boolean,
+});
+
+const { t } = useI18n();
+
+const expand = ref(false);
+
+const formattedDate = computed(() =>
+  new Date(
+    /^\d+$/.test(props.date) ? Number(props.date) * 1000 : props.date
+  ).toLocaleDateString("en-CA")
+);
+
+const formattedSize = computed(() => {
+  if (!props.size) return "";
+  const units = ["B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
+  const unit = Math.floor(Math.log(props.size) / Math.log(1024));
```

Comment:
1024-based units should use the IEC prefix ("KiB", "MiB" etc.). SI prefixes are 1000-based.

We can choose to use the unambiguous prefix because in practice those "beginners" will also fail to recognize that "KB" "MB" etc. are 1000-based, so readability for them wouldn't get hurt by either style.